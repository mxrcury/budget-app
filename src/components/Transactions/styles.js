import styled from 'styled-components'

export const Wrapper = styled.div`
    display:flex;
    justify-content:space-between;
    align-items:center;
    outline:1px solid rgb(125, 125, 125);
    padding:2px 7px;
    width:90%;
    margin:5px;
    margin-bottom:10px;
`
Wrapper.displayName = 'TransactionWrapper'

export const Icon = styled.div`
`

Icon.displayName = 'StarIcon'