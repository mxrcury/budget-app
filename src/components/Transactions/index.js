import PropTypes from "prop-types";
import { Icon, Wrapper } from "./styles";
import React, { useContext } from 'react';
import AppContext from "../../providers/context";
import StarOutlineIcon from '@mui/icons-material/StarOutline';
import StarIcon from '@mui/icons-material/Star';

const Transactions = ({ transactions = [], onDelete, onStarClick }) => {

  const { state } = useContext(AppContext)

  return (
    <>
      {transactions.map((el, index) => (
        <Wrapper key={`${el}_${index}`} style={{background:Number(el.value) < 0 ? 'rgb(175, 175, 175)' : 'white'}}>
          <Icon
            onClick={() => {
              onStarClick(el.id);
            }}
          >
            {el.isStarred ? (
              <StarIcon sx={{ cursor: "pointer", color: "orange" }} />
            ) : (
              <StarOutlineIcon sx={{ cursor: "pointer" }} />
            )}
          </Icon>

          <div>
            {Number(el.value).toFixed(2)} {state.currency}
          </div>
          <div> {el.comment}</div>
          <div>{el.date}</div>
          <button onClick={() => onDelete(el.id)}>Delete</button>
        </Wrapper>
      ))}
    </>
  );
}

Transactions.propTypes = {
  transactions: PropTypes.arrayOf(PropTypes.object),
  onDelete:PropTypes.func,
  onStarClick:PropTypes.func,
};

export default Transactions;
