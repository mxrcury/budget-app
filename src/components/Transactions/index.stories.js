import Transactions from "."

export default{
    title:'Main/Transactions',
    component:Transactions
}

const Template = (args) => <Transactions {...args} />


export const Income = Template.bind({});
Income.args = {
    transactions:[{
        value:'income',
        date:312321312321
    }]
}