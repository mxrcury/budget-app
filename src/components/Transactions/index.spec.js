import React from 'react'
import Transactions from '.'
import {shallow} from 'enzyme'


describe('Transaction component',()=>{
    it('should show transaction',()=>{
        const sut = shallow(<Transactions/>)
        
        expect(sut).toMatchSnapshot()
    })
})