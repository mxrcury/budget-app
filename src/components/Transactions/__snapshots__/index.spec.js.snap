// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should show transaction 1`] = `ShallowWrapper {}`;
